core/capabilities/ccip/ccip_integration_tests/usdcreader/usdcreader_test.go

```diff
@@ -201,7 +207,147 @@ func Test_USDCReader_MessageHashes(t *testing.T) {
+					sourceChain: ts.reader,
+				})
+			require.NoError(b, err)
+
+			// Populate the database with the specified number of logs
+			populateDatabase(b, ts, sourceChain, sourceDomainCCTP, destDomainCCTP, tc.startNonce, tc.msgCount, finalityDepth)
+
+			// Create a map of tokens to query for, with the specified tokenCount
+			tokens := make(map[reader.MessageTokenID]cciptypes.RampTokenAmount)
+			for i := 1; i <= tc.tokenCount; i++ {
+				tokens[reader.NewMessageTokenID(cciptypes.SeqNum(i), 1)] = cciptypes.RampTokenAmount{
+					ExtraData: reader.NewSourceTokenDataPayload(tc.startNonce+uint64(i), sourceDomainCCTP).ToBytes(),
+				}
+			}
+
+			b.ResetTimer()
+
+			for i := 0; i < b.N; i++ {
+				hashes, err := usdcReader.MessageHashes(ctx, sourceChain, destChain, tokens)
+				require.NoError(b, err)
+				require.Len(b, hashes, tc.tokenCount) // Ensure the number of matches is as expected
+			}
+		})
+	}
+}
 
+func populateDatabase(b *testing.B,
+	testEnv *testSetupData,
+	source cciptypes.ChainSelector,
+	sourceDomainCCTP uint32,
+	destDomainCCTP uint32,
+	startNonce uint64,
+	numOfMessages int,
+	finalityDepth int) {
+	ctx := testutils.Context(b)
+
+	abi, err := usdc_reader_tester.USDCReaderTesterMetaData.GetAbi()
+	require.NoError(b, err)
+
+	var logs []logpoller.Log
+	messageSentEventSig := abi.Events["MessageSent"].ID
+	require.NoError(b, err)
+	messageTransmitterAddress := testEnv.contractAddr
+
+	for i := 0; i < numOfMessages; i++ {
+		nonce := startNonce + uint64(i)
+
+		// Pack the message following exact CCTP format
+		var buf []byte
+		// _msgVersion (uint32)
+		buf = binary.BigEndian.AppendUint32(buf, reader.CCTPMessageVersion)
+		// _msgSourceDomain (uint32)
+		buf = binary.BigEndian.AppendUint32(buf, sourceDomainCCTP)
+		// _msgDestinationDomain (uint32)
+		buf = binary.BigEndian.AppendUint32(buf, destDomainCCTP)
+		// _msgNonce (uint64)
+		buf = binary.BigEndian.AppendUint64(buf, nonce)
+		// First 12 bytes of the sender address are always empty for EVM
+		senderBytes := [12]byte{}
+		buf = append(buf, senderBytes[:]...)
+
+		// Convert to 32 bytes
+		var message [32]byte
+		copy(message[:], buf)
+
+		// Encode the data following Solidity's encoding for bytes
+		data := make([]byte, 0)
+
+		// This says "actual data starts 32 bytes from the beginning"
+		offsetBytes := make([]byte, 32)
+		binary.BigEndian.PutUint64(offsetBytes[24:], 32)
+		data = append(data, offsetBytes...)
+
+		// This is the length of our message
+		lengthBytes := make([]byte, 32)
+		binary.BigEndian.PutUint64(lengthBytes[24:], uint64(len(message)))
+		data = append(data, lengthBytes...)
+
+		// Add message
+		data = append(data, message[:]...)
+
+		// Create topics array with just the event signature
+		topics := [][]byte{
+			messageSentEventSig[:], // Topic[0] is event signature
+		}
+
+		// Create log entry
+		logs = append(logs, logpoller.Log{
+			EvmChainId:     ubig.New(big.NewInt(int64(uint64(source)))),
```

Comment:
hmm, is that right? We should use chain_id, not the chain selectors which is stored in `source`, right?

Reply:
The simulated backend is created with the source chain. The only part we are using the `ChainID` is for the transactor creation (as we don't have any other choice here, geth is complaining about it).